### ISSUE TITLE
Shouldn't panic on missing event fields

### ISSUE THREAD

**Issue by ara4n** [tdfischer/pto#50]
In accordance with Postel's law, it'd be good to be more forgiving of missing event fields (related to #48).  Currently any `mjson::path()` that fails to find a path panics entirely with `panic!("Could not find {} in {} (lost at {})", path, json.pretty(), p)`.  I obviously don't know any rust, but I'm assuming this would better be a Result rather than a panic so it can be trapped nicely in `from_room_json` and similar, rather than having to do the partial solution I've done in https://github.com/tdfischer/pto/pull/49 for the specific case of redactions.

**Comment by tdfischer** [comment]
Yeah, using the builtin json path functions would be a much loved improvement. The mjson functions prevent the need for a billion unwrap().unwrap().unwrap() when the majority of event packets seem to work out fine.

Redoing it with more thorough error checking would make the code here explode in complexity, a problem I've wrestled with cleanly solving for some time. Probably about time for a macro.
